from PyQt5 import QtCore, QtGui, QtWidgets


class Ui_subject_allotment(object):
    def setupUi(self, subject_allotment):
        subject_allotment.setObjectName("subject_allotment")
        subject_allotment.resize(720, 480)
        icon = QtGui.QIcon()
        icon.addPixmap(QtGui.QPixmap("assets/FaceAttend2.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        subject_allotment.setWindowIcon(icon)
        self.gridLayout_4 = QtWidgets.QGridLayout(subject_allotment)
        self.gridLayout_4.setObjectName("gridLayout_4")
        self.list_widget_available = QtWidgets.QListWidget(subject_allotment)
        self.list_widget_available.setObjectName("list_widget_available")
        self.gridLayout_4.addWidget(self.list_widget_available, 6, 0, 11, 1)
        self.button_remove = QtWidgets.QPushButton(subject_allotment)
        self.button_remove.setText("")
        icon1 = QtGui.QIcon()
        icon1.addPixmap(QtGui.QPixmap("assets/left.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.button_remove.setIcon(icon1)
        self.button_remove.setIconSize(QtCore.QSize(32, 32))
        self.button_remove.setFlat(False)
        self.button_remove.setObjectName("button_remove")
        self.gridLayout_4.addWidget(self.button_remove, 12, 1, 1, 1, QtCore.Qt.AlignHCenter)
        self.button_add = QtWidgets.QPushButton(subject_allotment)
        self.button_add.setText("")
        icon2 = QtGui.QIcon()
        icon2.addPixmap(QtGui.QPixmap("assets/right.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.button_add.setIcon(icon2)
        self.button_add.setIconSize(QtCore.QSize(32, 32))
        self.button_add.setFlat(False)
        self.button_add.setObjectName("button_add")
        self.gridLayout_4.addWidget(self.button_add, 10, 1, 1, 1, QtCore.Qt.AlignHCenter)
        self.button_select = QtWidgets.QPushButton(subject_allotment)
        self.button_select.setText("")
        icon3 = QtGui.QIcon()
        icon3.addPixmap(QtGui.QPixmap("assets/select.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.button_select.setIcon(icon3)
        self.button_select.setIconSize(QtCore.QSize(32, 32))
        self.button_select.setFlat(False)
        self.button_select.setObjectName("button_select")
        self.gridLayout_4.addWidget(self.button_select, 4, 1, 1, 1, QtCore.Qt.AlignHCenter)
        self.label_placeholder = QtWidgets.QLabel(subject_allotment)
        self.label_placeholder.setObjectName("label_placeholder")
        self.gridLayout_4.addWidget(self.label_placeholder, 2, 0, 1, 3, QtCore.Qt.AlignHCenter|QtCore.Qt.AlignVCenter)
        self.list_widget_allotted = QtWidgets.QListWidget(subject_allotment)
        self.list_widget_allotted.setObjectName("list_widget_allotted")
        self.gridLayout_4.addWidget(self.list_widget_allotted, 6, 2, 11, 1)
        self.label_teachers_icon = QtWidgets.QLabel(subject_allotment)
        self.label_teachers_icon.setText("")
        self.label_teachers_icon.setPixmap(QtGui.QPixmap("assets/group.png"))
        self.label_teachers_icon.setObjectName("label_teachers_icon")
        self.gridLayout_4.addWidget(self.label_teachers_icon, 0, 1, 1, 1)
        self.label_available = QtWidgets.QLabel(subject_allotment)
        self.label_available.setObjectName("label_available")
        self.gridLayout_4.addWidget(self.label_available, 5, 0, 1, 1, QtCore.Qt.AlignHCenter)
        self.label_allotted = QtWidgets.QLabel(subject_allotment)
        self.label_allotted.setObjectName("label_allotted")
        self.gridLayout_4.addWidget(self.label_allotted, 5, 2, 1, 1, QtCore.Qt.AlignHCenter)
        self.button_update = QtWidgets.QPushButton(subject_allotment)
        self.button_update.setText("")
        icon4 = QtGui.QIcon()
        icon4.addPixmap(QtGui.QPixmap("assets/tick.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.button_update.setIcon(icon4)
        self.button_update.setIconSize(QtCore.QSize(32, 32))
        self.button_update.setFlat(False)
        self.button_update.setObjectName("button_update")
        self.gridLayout_4.addWidget(self.button_update, 18, 0, 1, 3, QtCore.Qt.AlignHCenter)
        self.label_spacer_bottom = QtWidgets.QLabel(subject_allotment)
        self.label_spacer_bottom.setText("")
        self.label_spacer_bottom.setObjectName("label_spacer_bottom")
        self.gridLayout_4.addWidget(self.label_spacer_bottom, 15, 1, 1, 1)
        self.label_spacer_top = QtWidgets.QLabel(subject_allotment)
        self.label_spacer_top.setText("")
        self.label_spacer_top.setObjectName("label_spacer_top")
        self.gridLayout_4.addWidget(self.label_spacer_top, 9, 1, 1, 1)
        self.comboBox = QtWidgets.QComboBox(subject_allotment)
        self.comboBox.setObjectName("comboBox")
        self.gridLayout_4.addWidget(self.comboBox, 3, 0, 1, 3)

        self.retranslateUi(subject_allotment)
        QtCore.QMetaObject.connectSlotsByName(subject_allotment)
        subject_allotment.setTabOrder(self.comboBox, self.button_select)
        subject_allotment.setTabOrder(self.button_select, self.list_widget_available)
        subject_allotment.setTabOrder(self.list_widget_available, self.list_widget_allotted)
        subject_allotment.setTabOrder(self.list_widget_allotted, self.button_add)
        subject_allotment.setTabOrder(self.button_add, self.button_remove)
        subject_allotment.setTabOrder(self.button_remove, self.button_update)

    def retranslateUi(self, subject_allotment):
        _translate = QtCore.QCoreApplication.translate
        subject_allotment.setWindowTitle(_translate("subject_allotment", "Subject Allotment"))
        self.label_placeholder.setText(_translate("subject_allotment", "You can allot or withhold the subject(s) by selecting name of the teacher from the list below:"))
        self.label_available.setText(_translate("subject_allotment", "Available Subject(s)"))
        self.label_allotted.setText(_translate("subject_allotment", "Allotted Subject(s)"))


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    subject_allotment = QtWidgets.QWidget()
    ui = Ui_subject_allotment()
    ui.setupUi(subject_allotment)
    subject_allotment.show()
    sys.exit(app.exec_())
